src/xdocs/checks/javadoc/javadocleadingasteriskalign.xml

```diff
@@ -0,0 +1,257 @@
+     Tight-HTML Rules</a>.</td>
+              <td><a href="../../property_types.html#boolean">boolean</a></td>
+              <td><code>false</code></td>
+              <td>10.17</td>
+            </tr>
+          </table>
+        </div>
+      </subsection>
+
+      <subsection name="Examples" id="Examples">
+        <p id="Example1-config">
+          To configure the check:
+        </p>
+        <source>
+&lt;module name=&quot;Checker&quot;&gt;
+  &lt;module name=&quot;TreeWalker&quot;&gt;
+    &lt;module name=&quot;JavadocLeadingAsteriskAlign&quot; /&gt;
+  &lt;/module&gt;
+&lt;/module&gt;
+        </source>
+        <p id="Example1-code">
+          Example with correct alignment:
+        </p>
+        <source>
+/** Title
+ * Javadoc for class
+ */
+public class Example1 {
+
+  /** javadoc for instance variable. */
+  private int ball;
+
+  /**
+   * Javadoc for instance variable
+   */
+  private int age;
+
+  /**
+   Javadoc for method. */
+  private void foo() {}
+
+  /**
+    Javadoc for Constructor.
+    This javadoc is allowed because there is no leading asterisk.
+   */
+  public Example1() {}
+
+  /**
+   * Javadoc for enum.
+   */
+  private enum correctJavadocEnum {
+    /**
+     * Correct Indentation for leading asterisk */
+    ONE,
+
+    /**
+     This is allowed because there is no leading asterisk. */
+    TWO
+  }
+}
+        </source>
+
+        <p id="Example2-code">
+          Example with incorrect alignment:
+        </p>
+        <source>
+/** Title
+* Javadoc for class // violation
+    */ // violation
+public class Example2 {
+  /**
+    *  Javadoc for instance variable. // violation
+    */ // violation
+  private String name;
+
+  /**
+  *  Javadoc for method. // violation
+  */ // violation
+  private void foo() {}
+
+  /**
+   Javadoc for Constructor.
+*/ // violation
+  private Example2() {}
+
+  /**
+    * Javadoc for enum. // violation
+   */
+  private enum incorrectJavadocEnum {
+
+    /**
+    *  // violation
+     */
+    ONE,
+
+    /**
+        * Incorrect indentation for leading asterisk. */ // violation
+    TWO,
+
+    /**
+ *    // violation
+     */
+    THREE
+  }
+}
+        </source>
+
+        <p id="Example3-config">
+          To configure the check with <code>tabWidth</code> property:
+        </p>
+        <source>
+&lt;module name=&quot;Checker&quot;&gt;
+  &lt;module name=&quot;TreeWalker&quot;&gt;
+    &lt;module name=&quot;JavadocLeadingAsteriskAlign&quot;&gt;
+      &lt;property name=&quot;tabWidth&quot; value=&quot;2&quot;/&gt;
+    &lt;/module&gt;
+  &lt;/module&gt;
+&lt;/module&gt;
+        </source>
+        <p id="Example3-code">
+          Example:
+        </p>
+        <source>
+/**
+ * Example with `tabWidth` property.
+ * This example contains Tabs as well as Spaces.
+ */
+public class Example3 {
+  /** &lt;- Preceded with Tabs.
+   * &lt;- Preceded with Tabs &amp; Spaces.
+   */ // &lt;- Preceded with Tabs &amp; Spaces.
```

Comment:
Yeah I got your point. But I tried searching if we are using tabs in any our documentation examples but didn't find any of them. Even the `FileTabCharacter` check is converting it's examples' tabs into `double whitespaces` here:

https://github.com/checkstyle/checkstyle/blob/b1bbce36e3704d0a1af1f9ff0e57f9a0721d40ae/src/main/java/com/puppycrawl/tools/checkstyle/site/ExampleMacro.java#L94-L99


@nrmancuso I would like to hear your opinion on this. Should we keep `tabs` in our documentation examples? 

The [Example3](https://github.com/checkstyle/checkstyle/blob/dd39b32f36c619dc2ef46ca34ec21b327fc4684f/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocleadingasteriskalign/Example3.java) of the documentation shows the usage of `tabsWidth` property but it isn't actually using tabs because if we did, the javadoc comment's indentation will look weird as you can see here: https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/576ca0d_2024222212/checks/javadoc/javadocleadingasteriskalign.html#Example3-code

This was the old Example3 which actually contained tabs and in the documentation it seems that one tab is taking width of 8 whitespaces. So I thought it would be better if just use whitespaces in our Example3, that way the indentation will not look weird. Also There aren't any documentation examples which uses tabs, the `FileTabCharacter` is also converting its tabs into whitespaces for documentation examples as I mentioned above.